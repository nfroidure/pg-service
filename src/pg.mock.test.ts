import assert from 'assert';
import YError from 'yerror';
import initPGMock from './pg.mock';

describe('initPGMock', () => {
  test('should work', async () => {
    const pg = await initPGMock();

    assert(pg);
  });

  describe('pg.query', () => {
    test('should work', async () => {
      const { service: pg } = await initPGMock();

      pg.query.mockResultOnce({ rows: [], fields: {} });

      await pg.query('SELECT * FROM users WHERE id = $$userId', { userId: 1 });

      expect(pg.query.mock.calls).toMatchSnapshot();
    });

    test('should fail with lacking args', async () => {
      const { service: pg } = await initPGMock();

      pg.query.mockResultOnce({ rows: [], fields: {} });

      try {
        await pg.query('SELECT * FROM users WHERE id = $$userId', {});
        throw new YError('E_UNEXPECTED_SUCCESS');
      } catch (err) {
        expect({
          errorCode: err.code,
          errorParams: err.params,
        }).toMatchSnapshot();
      }
    });

    test('should fail with malformed query', async () => {
      const { service: pg } = await initPGMock();

      pg.query.mockResultOnce({ rows: [], fields: {} });

      try {
        await pg.query('SELECT * FROM', {});
        throw new YError('E_UNEXPECTED_SUCCESS');
      } catch (err) {
        expect({
          errorCode: err.code,
          errorParams: err.params,
        }).toMatchSnapshot();
      }
    });
  });

  describe('pg.queries', () => {
    test('should work', async () => {
      const { service: pg } = await initPGMock();

      pg.queries.mockResultOnce([{ rows: [], fields: {} }]);

      await pg.queries(
        [
          'SELECT * FROM users WHERE id = $$userId',
          'SELECT * FROM users WHERE id = $$userId',
        ],
        { userId: 1 },
      );

      expect(pg.queries.mock.calls).toMatchSnapshot();
    });

    test('should fail with lacking args', async () => {
      const { service: pg } = await initPGMock();

      pg.queries.mockResultOnce([{ rows: [], fields: {} }]);

      try {
        await pg.queries(['SELECT * FROM users WHERE id = $$userId'], {});
        throw new YError('E_UNEXPECTED_SUCCESS');
      } catch (err) {
        expect({
          errorCode: err.code,
          errorParams: err.params,
        }).toMatchSnapshot();
      }
    });
  });

  describe('pg.transaction', () => {
    test('should work', async () => {
      const { service: pg } = await initPGMock();

      pg.transaction.mockResultOnce([{ rows: [], fields: {} }]);

      await pg.transaction(
        [
          'SELECT * FROM users WHERE id = $$userId',
          'SELECT * FROM users WHERE id = $$userId',
        ],
        { userId: 1 },
      );

      expect(pg.transaction.mock.calls).toMatchSnapshot();
    });

    test('should fail with lacking args', async () => {
      const { service: pg } = await initPGMock();

      pg.transaction.mockResultOnce([{ rows: [], fields: {} }]);

      try {
        await pg.transaction(['SELECT * FROM users WHERE id = $$userId'], {});
        throw new YError('E_UNEXPECTED_SUCCESS');
      } catch (err) {
        expect({
          errorCode: err.code,
          errorParams: err.params,
        }).toMatchSnapshot();
      }
    });
  });
});
