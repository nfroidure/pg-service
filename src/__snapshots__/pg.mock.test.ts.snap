// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initPGMock pg.queries should fail with lacking args 1`] = `
Object {
  "errorCode": "E_PG_LACKING_ARG",
  "errorParams": Array [
    "SELECT * FROM users WHERE id = $$userId",
    Object {},
    "userId",
  ],
}
`;

exports[`initPGMock pg.queries should work 1`] = `
Array [
  Array [
    Array [
      "SELECT * FROM users WHERE id = $$userId",
      "SELECT * FROM users WHERE id = $$userId",
    ],
    Object {
      "userId": 1,
    },
  ],
]
`;

exports[`initPGMock pg.query should fail with lacking args 1`] = `
Object {
  "errorCode": "E_PG_LACKING_ARG",
  "errorParams": Array [
    "SELECT * FROM users WHERE id = $$userId",
    Object {},
    "userId",
  ],
}
`;

exports[`initPGMock pg.query should fail with malformed query 1`] = `
Object {
  "errorCode": "E_INVALID_QUERY",
  "errorParams": Array [
    "SELECT * FROM",
    [Error: syntax error at end of input],
  ],
}
`;

exports[`initPGMock pg.query should work 1`] = `
Array [
  Array [
    "SELECT * FROM users WHERE id = $$userId",
    Object {
      "userId": 1,
    },
  ],
]
`;

exports[`initPGMock pg.transaction should fail with lacking args 1`] = `
Object {
  "errorCode": "E_PG_LACKING_ARG",
  "errorParams": Array [
    "SELECT * FROM users WHERE id = $$userId",
    Object {},
    "userId",
  ],
}
`;

exports[`initPGMock pg.transaction should work 1`] = `
Array [
  Array [
    Array [
      "SELECT * FROM users WHERE id = $$userId",
      "SELECT * FROM users WHERE id = $$userId",
    ],
    Object {
      "userId": 1,
    },
  ],
]
`;
